indexer/indexer_tests/feature_names_test.cpp

```diff
@@ -1,39 +1,270 @@
 #include "testing/testing.hpp"
 
+#include "indexer/feature_utils.hpp"
 #include "indexer/features_vector.hpp"
 #include "indexer/index.hpp"
 #include "indexer/mwm_set.hpp"
 
 #include "platform/local_country_file.hpp"
+#include "platform/preferred_languages.hpp"
 
 using namespace platform;
 
 namespace
 {
-UNIT_TEST(GetFeatureNames)
+using StrUtf8 = StringUtf8Multilang;
+
+UNIT_TEST(GetPrefferedNames)
 {
-  LocalCountryFile localFile = LocalCountryFile::MakeForTesting("minsk-pass");
-
-  Index index;
-  auto result = index.RegisterMap(localFile);
-  TEST_EQUAL(result.second, MwmSet::RegResult::Success, ());
-
-  auto const & id = result.first;
-  MwmSet::MwmHandle handle = index.GetMwmHandleById(id);
-  TEST(handle.IsAlive(), ());
-
-  auto const * value = handle.GetValue<MwmValue>();
-  FeaturesVector fv(value->m_cont, value->GetHeader(), value->m_table.get());
-  string primary, secondary, readable;
-
-  fv.ForEach([&](FeatureType & ft, uint32_t /* index */)
-  {
-    ft.GetPreferredNames(primary, secondary);
-    if (!secondary.empty())
-    {
-      ft.GetReadableName(readable);
-      TEST_EQUAL(secondary, readable, ());
-    }
-  });
+  vector<string> const & mwmLanguages = {"de", "ko"};
+  feature::RegionData regionData;
+  regionData.SetLanguages(mwmLanguages);
+
+  int8_t deviceLang = StrUtf8::GetLangIndex("ru");
+  string primary, secondary;
+
+  // Remove method is not implemented for StringUtf8Multilang
+  unique_ptr<StrUtf8> src = make_unique<StrUtf8>();
```

Comment:
Why do you need all this stuff with unique_ptr at all? I guess it's much simpler to do so:
```c++
{
   StrUtf8 src;
   src.AddString(...);
   ...
}

{
  StrUtf8 src;
  ...
}
...
```